lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -485,14 +489,21 @@ private function getRealClass(string $class): string
     private function createDefaultProxyClassNameResolver(): void
     {
         $this->proxyClassNameResolver = new class implements ProxyClassNameResolver {
+            /**
+             * @template T of object
+             * @psalm-param class-string<Proxy<T>>|class-string<T> $className
```

Comment:
I've copied this because otherwise the `/** @psalm-var class-string<T> */` won't recognise the `T`.

Reply:
Seems like this is not enough, the tests are still failing

Reply:
Apparently [it should work](https://github.com/phpstan/phpstan/discussions/4802), I'll ignore for now.